Project.toml

```
@@ -1,14 +1,15 @@
 name = "DocumenterTools"
 uuid = "35a29f4d-8980-5a13-9543-d66fff28ecb8"
-version = "0.1.0"
+version = "0.1.1"
```

#22 was not a bugfix, so bumping the minor version would be more appropriate I think:

```suggestion
version = "0.2.0"
```

---

Pkg docs:
> While the semver specification says that all versions with a major version of 0 are incompatible with each other, we have made that choice that a version given as 0.a.b is considered compatible with 0.a.c if a != 0 and c >= b.

So unless it was breaking (I don't think it was?) I think we should go with 0.1.1.

---

It was non-breaking, but it was a functionality change (introduced a new method). My interpretation has always been that 0.x.y patch versions are only for bugfixes, i.e. no functionality has changed. in other words, semver (6) still applies:

> 6. Patch version Z (x.y.Z | x > 0) MUST be incremented if only backwards compatible bug fixes are introduced. A bug fix is defined as an internal change that fixes incorrect behavior.

And that has been the practice with Documenter. However, if it is a misinterpretation, then I don't mind changing the practice.

---

>  x > 0

---

Yes, but it's not clear to me whether it's the (6) or the (7) compatibility that applies in the Pkg exception. I always assumed it's (6).

---

Pkg treats `0.(X+1)` as breaking compared to `0.X`, but does not treat `0.X.(Y+1)` as breaking compared to `0.X.Y`.

---

Yes, but the question is whether code written for `0.X.(Y+1)` should also work with `0.X.Y`, which is the normal patch release promise.

---

For what it's worth though, I can see arguments either way.  On the one hand, it would be nice to be able to give a strong stability guarantee during the 0.x phase, but still be able to provide bugfixes. But on the other, having the minor releases clearly demarcating breaking releases can also be useful.

---

Oh right. I think we do patches and features in patch releases pre 1.0. It would be annoying if we had to do a breaking release for each new feature pre 1.0.

---

I guess my view here is shaped by the practice in core Julia (IIRC the 0.x patch versions were bugfixes only, with new features etc. all going into the next minor versions), and how we've handled Documenter's versions. I haven't really paid attention to how other packages have handled this.

But yeah, as there were no strong opinions about it on Slack, let's just go with 0.1.1 then. Minor 0.x versions are then for breaking releases and perhaps for releases where some major functionality has been added.

I would probably stay conservative with Documenter itself though since patch releases get deployed automatically. But happy to be more liberal with the extension repos. In any case, hopefully we'll be out the of 0.x era soon anyway :slightly_smiling_face: